### PR TITLE
[Feature] I-JEPA + MAE benchmarks on ImageNet

### PR DESCRIPTION
**Pull request by sami-bg**
## Description
Training currently in progress on ImageNet100
#382 #383 

## Checklist

- [ ] I have read the [**Contributing**](https://rbalestr-lab.github.io/stable-SSL.github.io/dev/contributing.html#) document.
- [ ] The documentation is up-to-date with the changes I made (check build artifacts).
- [ ] All tests passed, and additional code has been **covered with new tests**.
- [ ] I have added the PR to the [**RELEASES.rst**](../RELEASES.rst) file.

### CODE REVIEW

File: stable_pretraining/backbone/vit.py [hunk 1]
```diff
@@ -352,22 +356,30 @@ def _get_prefix_tokens(self, B: int) -> Optional[torch.Tensor]:
             tokens.append(self.vit.reg_token.expand(B, -1, -1))
         return torch.cat(tokens, dim=1) if tokens else None
 
-    def forward(self, images: torch.Tensor) -> MaskedEncoderOutput:
+    def forward(
+        self, images: torch.Tensor, apply_mask: bool | None = None
```
Comment (RandallBalestriero) [inline stable_pretraining/backbone/vit.py]: what does it give us extra to have this input? do you have in mind a case where you are in training but don't want to apply mask or apply mask in eval? curious to know the use-case

File: benchmarks/imagenet10/auto_probe_inet10.py [hunk 2]
```diff
@@ -0,0 +1,216 @@
+"""Offline linear probe evaluation for I-JEPA checkpoints on ImageNet-10.
```
Comment (RandallBalestriero) [inline benchmarks/imagenet10/auto_probe_inet10.py]: I think overall we could cut down the # of lines by at least 2 given all the utilities we provide! 
First I think it would be much easier to follow by not having any helper function and doing it all in a main() function
Second let's try to not implement anything but use out of the box our callback for probing etc
Third, why do you need to handle ckpt etc yourself? if there is an interesting use-case we should push to our utilities, if it is too specific to your setup then I would remove from this general file

**Review by RandallBalestriero — CHANGES_REQUESTED** [review]
